src/Compilers/CSharp/Test/Semantic/Semantics/InterceptorsTests.cs

```diff
@@ -968,6 +1377,37 @@ static class D
             );
     }
 
+    [Fact]
+    public void InterceptableGeneric_03()
+    {
+        var source = """
+            using System.Runtime.CompilerServices;
+            using System;
+
+            class C
+            {
+                [Interceptable]
+                public static void InterceptableMethod<T>(T t) where T : class => throw null!;
+            }
+
+            static class Program
+            {
+                public static void Main()
+                {
+                    C.InterceptableMethod<string>("1");
+                }
+            }
+
+            static class D
+            {
+                [InterceptsLocation("Program.cs", 14, 11)]
+                public static void Interceptor1(string s) { Console.Write(s); }
+            }
+            """;
+        var verifier = CompileAndVerify(new[] { (source, "Program.cs"), s_attributesSource }, expectedOutput: "1");
+        verifier.VerifyDiagnostics();
+    }
```

Comment:
>} [](http://example.com/codeflow?start=4&length=1)

Is it possible to intercept a method call that uses method type arguments?
```csharp
class C
{
    [Interceptable]
    static void F1<T>(T t) { }
    
    static void F2<T>(T t)
    {
        F1(t); // intercept this?
    }
}
```

Reply:
Hmm... the idea is that `T` in `F2` is the type argument to `T` in `F1`, so how could you possibly intercept that?

I think the only way we could is by permitting an interceptor with the same arity and constraints as the interceptable method, and "reusing" the type arguments from the original call.

I'll add a test and note it as a limitation in the spec.